aten/src/ATen/native/cpu/UnaryOpsKernel.cpp

```diff
@@ -302,6 +302,17 @@ static void sgn_kernel(TensorIterator& iter){
   });
 }
 
+static void sinc_kernel(TensorIterator& iter) {
+  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND1(kBFloat16, iter.common_dtype(), "sinc_cpu", [&]() {
+    cpu_kernel(
+        iter,
+        [=](scalar_t a) -> scalar_t {
+          scalar_t product = scalar_t(M_PI) * a;
```

Comment:
Do the check for the zero special-case before computing the product and add a comment explaining why it's special-cased.